recipes/libcudss/meta.yaml

```diff
@@ -0,0 +1,198 @@
+
+  - name: libcudss-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcudss" ~ somajor, max_pin="x.x.x") }}
+    files:
+      - lib/libcudss.so  # [linux]
+      - lib/cudss.lib  # [win]
+      - include/cudss*
+      - lib/cmake/cudss
+    requirements:
+      host:
+        - {{ pin_subpackage("libcudss" ~ somajor, exact=True) }}
+      run:
+        - {{ pin_subpackage("libcudss" ~ somajor, exact=True) }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      files:
+        - test
+      requires:    # [build_platform == target_platform]
+        - {{ compiler("c") }}  # [build_platform == target_platform]
+        - {{ compiler("cxx") }}  # [build_platform == target_platform]
+        - {{ compiler('cuda') }}  # [build_platform == target_platform]
+        - {{ stdlib("c") }}  # [build_platform == target_platform]
+        - cmake   # [build_platform == target_platform]
+        - ninja  # [build_platform == target_platform]
+      commands:
+        - cmake ${CMAKE_ARGS} -GNinja test  # [build_platform == target_platform]
+        - cmake --build .  # [build_platform == target_platform]
+        - test -f $PREFIX/include/cudss.h  # [linux]
+        - test -f $PREFIX/include/cudss_distributed_interface.h  #[linux]
+        - test -L $PREFIX/lib/libcudss.so  # [linux]
+        - test ! -f $PREFIX/lib/libcudss_static.a  # [linux]
+        - if not exist %LIBRARY_LIB%\\cudss.lib exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\cudss.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\cudss_distributed_interface.h exit 1  # [win]
+
+# loadable modules; optional and only needed at runtime
+
+  - name: libcudss-commlayer-nccl
+    build:
+      skip: True  # [not linux]
+    requirements:
+      run:
+        - {{ pin_subpackage("libcudss-commlayer-nccl" ~ somajor, exact=True) }}
+    about:
+      summary: >-
+        The NCCL communications layer module for NVIDIA cuDSS
+
+  - name: libcudss-commlayer-nccl{{ somajor }}
+    build:
+      skip: True  # [not linux]
+    files:
+      - lib/libcudss_commlayer_nccl.so.*  # [linux]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - {{ stdlib("c") }}
+      host:
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+        # FIXME: Get actual version ranges from kiril
+        - nccl >=2.0,<3  # [linux]
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcudss_commlayer_nccl.so.{{ soversion }}  # [linux]
+        - test -L $PREFIX/lib/libcudss_commlayer_nccl.so.{{ somajor }}  # [linux]
+    about:
+      summary: >-
+        The NCCL communications layer module for NVIDIA cuDSS
+
+  - name: libcudss-commlayer-mpi
```

Comment:
I've learned recently about conda having an mpi lock ( = single mpi per environment). Will `libcudss-commlayer-mpi` be takling advantage of that? E.g., uf the user first installs an incompatible MPI, will we be able to report smth if the user attempt later to install `libcudss-commlayer-mpi`?

Reply:
The `libcudss-commlayer-mpi` package requires openmpi, so conda will switch to openmpi if another mpi is installed in the environment or it will raise an error because it is not able to find an mpi that is compatible with all the requested packages.